### PR TITLE
1.2 Authentication & authorization

### PR DESCRIPTION
**Pull request by alhimick-prog**
# Description
1.2 Authentication & authorization
https://github.com/alhimick-prog/cinematichub/projects/1#card-80434715

## Type of change

- [x] New feature 

# Checklist:

- [x] I have performed a self-review of my own code

### CODE REVIEW

**Review by ShrewHools — COMMENTED** [review]
commit name is not up to standard

File: db/migrate/20220417160703_devise_create_users.rb [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+      ## Recoverable
+      t.string   :reset_password_token
+      t.datetime :reset_password_sent_at
+
+      ## Rememberable
+      t.datetime :remember_created_at
+
+      t.string :name, null: false
+      t.string :nickname
+      t.integer :role, default: 2
```
Comment (ShrewHools) [inline db/migrate/20220417160703_devise_create_users.rb:19]: what do you need this field for now?

File: config/routes.rb [hunk 2]
```diff
@@ -1,3 +1,4 @@
 Rails.application.routes.draw do
-  # For details on the DSL available within this file, see https://guides.rubyonrails.org/routing.html
+  root 'home#index'
+  devise_for :users
```
Comment (ShrewHools) [inline config/routes.rb:2]: external routes are written first.
then application routes

File: app/views/users/unlocks/new.html.erb [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+<%= simple_form_for(resource, as: resource_name, url: unlock_path(resource_name), html: { method: :post }) do |f| %>
+  <%= f.error_notification %>
+  <%= f.full_error :unlock_token %>
+
+  <div class="form-inputs">
+    <%= f.input :email,
+                required: true,
+                autofocus: true,
+                input_html: { autocomplete: "email" } %>
```
Comment (ShrewHools) [inline app/views/users/unlocks/new.html.erb]: single quotes. Review all views

File: app/views/users/shared/_links.html.erb [hunk 4]
```diff
@@ -0,0 +1,25 @@
+<%- if controller_name != 'sessions' %>
```
Comment (ShrewHools) [inline app/views/users/shared/_links.html.erb:1]: read about the difference between `<%- %>` and `<% %>`

File: app/views/users/registrations/new.html.erb [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+                input_html: { autocomplete: "new-password" } %>
+    <%= f.input :password_confirmation,
+                required: true,
+                input_html: { autocomplete: "new-password" } %>
+    <%= f.input :name,
+                require: true %>
+    <%= f.input :nickname,
+                require: false %>
+    <%= f.input :birthday,
+                require: false %>
```
Comment (ShrewHools) [inline app/views/users/registrations/new.html.erb]: what is the newline for?

File: app/views/users/shared/_links.html.erb (hunk 4 above)
Comment (alhimick-prog) [reply app/views/users/shared/_links.html.erb:1]: > read about the difference between `<%- %>` and `<% %>`

Do `<%-` and `-%>` ignore newlines?

**Review by ShrewHools — APPROVED** [review]

**Pull request merged**